lib/active_triples/persistence_strategies/repository_strategy.rb

```diff
@@ -0,0 +1,78 @@
+module ActiveTriples
+  ##
+  # Persistence strategy for projecting `RDFSource` to `RDF::Repositories`.
+  class RepositoryStrategy
+    include PersistenceStrategy
+
+    # @!attribute [r] obj
+    #   the source to persist with this strategy
+    attr_reader :obj
+
+    ##
+    # @param obj [RDFSource, RDF::Enumerable] the `RDFSource` (or other
+    #   `RDF::Enumerable` to persist with the strategy.
+    def initialize(obj)
+      @obj = obj
+    end
+
+    ##
+    # Clear out any old assertions in the repository about this node or statement
+    # thus preparing to receive the updated assertions.
+    def erase_old_resource
+      if obj.node?
+        repository.statements.each do |statement|
+          repository.send(:delete_statement, statement) if
+            statement.subject == obj
+        end
+      else
+        repository.delete [obj.to_term, nil, nil]
+      end
+    end
+
+    ##
+    # Persists the object to the repository
+    #
+    # @return [Boolean]
+    def persist!
+      erase_old_resource
+      repository << obj
+      @persisted = true
+    end
+
+    ##
+    # Repopulates the graph from the repository.
+    #
+    # @return [Boolean]
+    def reload
+      obj << repository.query(subject: obj)
+      @persisted = true unless obj.empty?
+      true
+    end
+
+    ##
+    # @return [RDF::Repository] The RDF::Repository that the object will project
+    #   itself on when persisting.
+    def repository
+      @repository ||= set_repository
+    end
+
+    private
+
+      ##
+      # Finds an appropriate repository from the calling object's configuration.
+      # If no repository is configured, builds an ephemeral in-memory
+      # repository and 'persists' there.
+      #
+      # @todo find a way to move this logic out (PersistenceStrategyBuilder?).
```

Comment:
We should have a conversation about the trade-offs between `Builder` and `Factory`.  Either way, this might be an opportunity to address your concerns from https://github.com/ActiveTriples/ActiveTriples/pull/124#discussion_r30912399.

My inclination is to leave this as a `@todo`, for now.  Agreed?